North-Seattle-College/ad440-winter2021-thursday-repo
# Automate Alert Rule Creation

To view existing Alert Rules:
`Get-AzMetricAlertRuleV2 -ResourceGroupName nsc-functions-team1`

**Overview**
This PR creates a script that will create a new Alert Rule to check for 5xx response codes every 5 minutes, and send an alert if the count exceeds 30.
The script is set up so that only the first 5 parameters are required, and it will default to creating the described Rule, but any of the parameters can be added into the original script call to overwrite defaults.
This PR also edits parameter names in create_vnet.ps1 to conform to agreed-upon standards.

**To Test**
Navigate to /Automation/alert
`.\create_alert.ps1 -TenantId <tenant id> -SubscriptionId <subscription id> -SPApplicationId <service principal app id> -SPSecret <service principal secret> -AlertName <alert name>`

*Delete Resource*
Go to the Azure Portal and search for Function App - select `nsc-functionsapp-team1`

![image](https://user-images.githubusercontent.com/57076614/106548868-a6d0c700-64c4-11eb-84d9-582ba9ec5bac.png)

Scroll to the Alert blade on the left, then click on "Manage alert rules"
Observe that (unless you used custom settings) the conditions for your new rule are the same as the existing http-5xx-alert
Delete your newly created rule to save on resources : )
